ThatDefaultProject/Assets/ThatDefaultProject/Scripts/MaterialFlicker.cs

```diff
@@ -0,0 +1,43 @@
+using System.Collections;
+using UnityEngine;
+
+namespace that
+{
+    public class MaterialFlicker : MonoBehaviour
+    {
+        [SerializeField] private SkinnedMeshRenderer _skinnedMeshRenderer;
```

Comment:
Idea: cache a Renderer instead of a SkinnedMeshRenderer. Renderer is the base class of both SkinnedMeshRenderer and MeshRenderer. It also has a material property, so your code will work perfectly fine while keeping any renderer in mind.

Reply:
Extension of this: why not an array of Renderers?